lua/randomat2/cl_events/cl_poker_vgui.lua

```diff
@@ -0,0 +1,942 @@
+--// Logan Christianson
+
+local LoganPanel = {}
+LoganPanel.Font = "Trebuchet22"
+
+function LoganPanel:SetFont(newFont)
+    self.Font = newFont
+end
+
+-- Don't register this, it's used as a base for other vgui elements
+
+local LoganButton = table.Copy(LoganPanel)
+LoganButton.IsHover = false
+LoganButton.Disabled = false
+
+function LoganButton:SetDisabled(newDisabled)
+    self.Disabled = newDisabled
+end
+
+function LoganButton:SetHover(newHover)
+    self.IsHover = newHover
+end
+
+function LoganButton:OnCursorEntered()
+    if not self.Disabled then
+        self.IsHover = true
+    end
+end
+
+function LoganButton:OnCursorExited()
+    self.IsHover = false
+end
+
+function LoganButton:CustomDoClick()
+end
+
+function LoganButton:DoClick()
+    if not self.Disabled then
+        self:CustomDoClick()
+        surface.PlaySound("ui/buttonclick.wav")
+    end
+end
+
+vgui.Register("Poker_Button", LoganButton, "DButton")
+
+local PlayerCard = table.Copy(LoganPanel)
+PlayerCard.BetIcon = Material("vgui/ttt/randomats/poker/poker_hand.png")
+PlayerCard.Player = nil
+PlayerCard.Font = "cool_small"
+PlayerCard.NoPlayerName = "No Player!"
+PlayerCard.ActionText = ""
+PlayerCard.BetText = ""
+PlayerCard.BlindStatus = ""
+PlayerCard.Bet = 0
+PlayerCard.NameWidth = 0
+PlayerCard.NameHeight = 0
+PlayerCard.ActionWidth = 0
+PlayerCard.ActionHeight = 0
+PlayerCard.BetWidth = 0
+PlayerCard.BetHeight = 0
+PlayerCard.IsFolded = false
+PlayerCard.IsBetting = false
+
+function PlayerCard:SetFont(newfont)
+    self.Font = newFont
+
+    if self.Player then
+        self:CalculateNameSize()
+    end
+end
+
+function PlayerCard:SetPlayer(ply)
+    if ply then
+        self.Player = ply:Nick()
+
+        if string.len(self.Player) > 12 then
```

Comment:
Might need to do this to support weird characters in player names
```suggestion
        if utf8.len(self.Player) > 12 then
```

Reply:
Didn't even consider this a possibility